docs: set up top-level Installation section (#1162)
keptn/lifecycle-toolkit: docs/content/en/docs/install/upgrade.md

---
title: Upgrade
description: How to upgrade to the latest version of the Lifecycle Toolkit
icon: concepts
layout: quickstart
weight: 45
hidechildren: false # this flag hides all sub-pages in the sidebar-multicard.html
---

To upgrade from the previous version of the Keptn Lifecycle Toolkit to the latest,
run the same command sequence used to install KLT:

```shell
helm repo add klt https://charts.lifecycle.keptn.sh
helm repo update

> helm upgrade --install keptn klt/klt \

Reviewer: did we test this? AFAIK upgrade may fail due to crds...

Reviewer: IMO this may be confusing: if I had a version installed via manifest and run this, helm will fail

Author: This iteration just copies the info from the existing docs into the new structure.  It definitely needs to be revisited -- other comments question whether we should even document how to install from a manifest or whether we should mention the cert-manager installation.  And we definitely need to tell people how to upgrade using the helm chart if they previously installed with the manifest.  Also probably need to discuss the cert-manager -- do you have to use our cert manager, can our cert-manager co-exist on the same cluster with another cert-manager that may be instaleld for Argo or Flux...  Lots of stuff that needs to be clarified and expanded.  This PR is just the first draft of Phase 1 -- we still have muc work to do but baby steps...

   -n keptn-lifecycle-toolkit-system --create-namespace --wait
```

Use the `--set` flag or download and edit the `values.yaml` file
to modify the configuration as discused on the
[Install the Lifecycle Toolkit](../install/) page.

> ## Migrate from V 0.6.0 to V 0.7.0

Reviewer 2: I usually see `v` in lower case next to the number
```suggestion
## Migrate from v0.6.0 to v0.7.0
```

> Keptn Lifecycle Toolkit Version 0.7.0

Reviewer 3: ```suggestion
Keptn Lifecycle Toolkit Version v0.7.0
```

introduces the `metrics-operator`,
which is now separate from the `lifecycle-operator`.
Some functionality and behavior has been moved, changed, or renamed.

> Specifically, the `KeptnMetricProvider` CRD replaces

Reviewer 3: ```suggestion
Specifically, the `KeptnMetricsProvider` CRD replaces
```

the now-deprecated `KeptnEvaluationProvider` CRD.
Consequently, you must manually migrate the existing functionality
to the `KeptnMetricsProvider` CRD.

> Execute the following external bash script to do this:

Reviewer: should this be run before helm upgrade?

Author: Good question.  But I think we can discuss this and fix it in the next iteration, right?  I also need to create the reference page for the yaml file for the KeptnEvaluationProvider CRD and I expect cross-refs both to and from that page.

> ```sh

Reviewer 3: ```suggestion
```shell
```

curl -sL https://raw.githubusercontent.com/keptn/lifecycle-toolkit/main/.github/scripts/keptnevaluationprovider_migrator.sh | bash
```

This fetches and migrates all `KeptnEvaluationProvider` CRs

> for the cluster at which your `kubecon` text is pointing.

Reviewer 3: ```suggestion
for the cluster at which your kubernetes  context is pointing.
```
